src/chain_spec.rs

```diff
@@ -352,6 +361,15 @@ fn testnet_genesis(
 			default_paid_membership_fee: 100u128,
 			first_member_id: 1,
 		}),
+		forum: Some(ForumConfig {
+			category_by_id: vec![],
+			thread_by_id: vec![],
+			post_by_id: vec![],
+			next_category_id: 1,
```

Comment:
Shouldn't all these `next` fields have initial value 0?, this way the first object has id 0, which is what we want?

Reply:
Actually, we want the first id = 1. It's a common practice among different applications that id should not be 0. Here I refer to RDBMS and other storage systems. There never assume that id == 0. And in some systems, 0 could mean that id is not set.

Additionally, if you take a look at our other modules (like the membership) where we use ids, you can find that we use a slightly different but more flexible approach: we have `first_member_id` and `next_member_id`.

Reply:
One more thing: `id != index` (id doesn't mean index)